test/torch_test_meta.py

```diff
@@ -159,60 +159,57 @@
-    'test_boolean_indexing_weirdness_tensors',  # runtime error
+    'test_broaderrors_indexing',  # expecting a different runtime error
+    'test_broadcast_subspace',  # XLA doesn't support arange with negative step
+    'test_boolean_shape_mismatch',  # expecting a different runtime error
+    'test_boolean_indexing_weirdness',  # expecting a different runtime error
+    'test_boolean_indexing_weirdness_tensors',  # expecting a different runtime error
 
     # test_nn.py
     # TestNNDeviceType
-    'test_embedding_backward',  # uses sparse
+    'test_embedding_backward',  # sparse
     'test_embedding_dense_grad',  # slow
-    'test_EmbeddingBag_per_sample_weights_and_new_offsets', # wrong result
-    'test_batchnorm_grad',
-    'test_gumbel_softmax_xla',
-    'test_rnn_retain_variables',
-    'test_pool_invalid_size',  # doesn't raise
-    'test_nonlinearity_propagate_nan',
+    'test_EmbeddingBag_per_sample_weights_and_new_offsets', # UndefinedTensorImpl::_singleton
+    'test_batchnorm_grad', # UndefinedTensorImpl::_singleton
+    'test_pool_invalid_size',  # expecting a different runtime error
+    'test_nonlinearity_propagate_nan', # relu6 with a nan tensor returns a tensor([0.]) instead of a nan tensor
     'test_InstanceNorm3d_general',  # precision (1e-2)
     'test_InstanceNorm2d_general',  # precision (1e-2)
     'test_InstanceNorm1d_general',  # precision (1e-2)
-    'test_EmbeddingBag_per_sample_weights_failures',  # runtime error
-    'test_variable_sequence',  # value error
-    'test_embedding_bag_device',  # runtime error
-    'test_embedding_bag_device',  # runtime error
-    'test_batchnorm_eval',  # runtime error
+    'test_EmbeddingBag_per_sample_weights_failures',  # expecting a different runtime error
+    'test_variable_sequence',  # PackedSequence batch_sizes.device.type should be CPU but is XLA
+    'test_embedding_bag_device',  # UndefinedTensorImpl::_singleton
+    'test_batchnorm_eval',  # UndefinedTensorImpl::_singleton
     'test_MaxPool2d_indices',  # lowering
     'test_MaxPool1d_indices',  # lowering
     'test_EmbeddingBag_per_sample_weights_and_offsets',  # runtime error
     'test_EmbeddingBag_per_sample_weights_and_no_offsets',  # runtime error
-    'test_nll_loss_empty_tensor_reduction_mean',  # floating point division 0 by 0
-    'test_softshrink_negative',  # runtime error
+    'test_nll_loss_empty_tensor_reduction_mean',  # floating point division 0 by 0, expecting nan but get 0
 
     # test_type_promotion.py
     # TestTypePromotion
-    'test_many_promotions',
-    'test_inplace',
-    'test_indexing',
-    'test_booleans',  # not less than or equal to 0.001
-    'test_alternate_result',
-    'test_alpha_mismatch',
-    'test_lt_with_type_promotion',  # runtime error
-    'test_half',  # half
+    'test_many_promotions', # stride
+    'test_inplace', # XLA allows adding int and double inplace
+    'test_indexing', # XLA allows int to double type promotion
```

Comment:
Type promotion tests are good to fix, so that we get exact same behavior as normal pytorch devices. ;)

Reply:
sounds good, I will look into that